superset-frontend/src/dashboard/util/getDashboardUrl.ts

```diff
@@ -48,12 +44,9 @@ export default function getDashboardUrl({
   if (standalone) {
     newSearchParams.set(URL_PARAMS.standalone.name, standalone.toString());
   }
-
-  if (dataMask) {
-    newSearchParams.set(
-      URL_PARAMS.nativeFilters.name,
-      rison.encode(replaceUndefinedByNull(dataMask)),
-    );
+  const dataMaskKey = getUrlParam(URL_PARAMS.nativeFiltersByCacheKey);
+  if (dataMaskKey) {
+    newSearchParams.set(URL_PARAMS.nativeFilters.name, dataMaskKey as string);
```

Comment:
I think we should try to avoid storing two types of state in a single key (see my comment on `URL_PARAMS` above). To make this backward compatible, I'd propose the following:
1. check if native filter key is provided. If so, fetch the native filter state from that and remove the native filter rison from the search params.
2. if no native filter key is provided, check if native filter rison parameter is provided. If so, parse that and initialize the filter state with it. Also, remove the native filter rison string from the URL when doing so to clean up the URL.
3. If neither 1) or 2) are true, initialize a new filter key with an empty state (I believe this is the current functionality).

Reply:
These case are already done in the DashboardPage and Filterbar.
dashboardpage
![Screen Shot 2021-12-16 at 4 40 58 PM](https://user-images.githubusercontent.com/17326228/146469714-31a9b494-b712-4b75-8b87-e5d58dd63a0b.png)
filterbar
![Screen Shot 2021-12-16 at 4 41 39 PM](https://user-images.githubusercontent.com/17326228/146469765-8afe7ec8-d447-40bd-b651-dc33110a9275.png)

Reply:
1-3 are now done.